packages/analytics-js-plugins/src/ketchConsentManager/index.ts

```diff
@@ -69,20 +69,44 @@ const KetchConsentManager = (): ExtensionPlugin => ({
       const allowedConsentIds = state.consents.data.value.allowedConsentIds as string[];
 
       try {
-        const { ketchConsentPurposes } = destConfig;
+        const { ketchConsentPurposes, consentManagement } = destConfig;
+        const matchPredicate = (consent: string) => allowedConsentIds.includes(consent);
 
-        // If the destination do not have this mapping events will be sent.
-        if (!ketchConsentPurposes || ketchConsentPurposes.length === 0) {
-          return true;
-        }
+        // Generic consent management
+        if (consentManagement) {
```

Comment:
is this code part duplicated between different provider plugins? shuld we separate in a reusable method?

Reply:
Yes, it repeated, but refactoring it into a reusable method would mean another plugin chunk; although a really small one.
WDYT?

Reply:
I see, ok leave it as is

Reply:
Cool. I went ahead and made the changes but stashed the code locally in case we need it in the future.